# Itération 2

#### Analyse de l’état de la diversité des arbres

Objectifs de l’activité

● Analyser la répartition des espèces d’arbres à Grenoble, l’évolution de cette diversité au cours du temps, comparaison avec les objectifs de la Métro en termes de diversification des espèces.

● Utilisation d’autres structures de données : Sets, Dict

In [2]:
import csv
import zipfile
import os
import itertools
import matplotlib.pyplot as plt

from pathlib import Path
from collections import Counter

#### Consignes

### ● Pour commencer, identifiez tous les genres botaniques représentés à Grenoble. Combien y en a-t-il ? Essayez plusieurs manières de résoudre ce problème.

In [3]:
with open('../data/trees.csv') as file:

    reader = csv.DictReader(file)
    genre_list = []

    for row in reader:
        genre = row['GENRE_BOTA'].split()
        genre_list.append(genre)

In [4]:
list_genres = list(itertools.chain.from_iterable(genre_list))
list_genres

['Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Ginkgo',
 'Salix',
 'Liriodendron',
 'Liriodendron',
 'Liriodendron',
 'Catalpa',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Pyrus',
 'Robinia',
 'Robinia',
 'Acer',
 'Acer',
 'Acer',
 'Zelkova',
 'Parrotia',
 'Fraxinus',
 'Fagus',
 'Fraxinus',
 'Acer',
 'Fraxinus',
 'Magnolia',
 'Taxus',
 'Fagus',
 'Fraxinus',
 'Alnus',
 'Alnus',
 'Picea',
 'Picea',
 'Taxus',


In [5]:
Counter(list_genres)

Counter({'Acer': 5276,
         'Platanus': 4663,
         'Pinus': 2118,
         'Tilia': 1770,
         'Fraxinus': 1506,
         'Carpinus': 1296,
         'Betula': 1247,
         'Prunus': 996,
         'Populus': 966,
         'Alnus': 826,
         'Liquidambar': 809,
         'Quercus': 677,
         'Liriodendron': 507,
         'Pyrus': 501,
         'Aesculus': 459,
         'Sophora': 426,
         'Magnolia': 409,
         'Cedrus': 407,
         'Chamaecyparis': 373,
         'Gleditsia': 352,
         'Cercis': 351,
         'Robinia': 287,
         'Koelreuteria': 271,
         'Fagus': 267,
         'Malus': 262,
         'Picea': 252,
         'Zelkova': 233,
         'Ulmus': 210,
         'Cupressus': 203,
         'Taxus': 202,
         'Celtis': 193,
         'Corylus': 174,
         'Salix': 143,
         'Ginkgo': 111,
         'Crataegus': 108,
         'Paulownia': 101,
         'Morus': 100,
         'Cupressocyparis': 100,
         'Amélanchier': 96,
     

In [6]:
counter_genres = Counter(list_genres)
len(counter_genres)

112

Il existe 112 genres botaniques différents à Grenoble

### ● Identifiez maintenant toutes les espèces différentes présentes à Grenoble. N’oubliez pas que les espèces sont une sous catégorie de genre botanique! (l’espèce est ‘Platanus Acerifolia’ et pas uniquement ‘Acerifolia’)

In [7]:
espèces_différentes = []

with open('../data/trees.csv', newline="") as file:

    reader = csv.DictReader(file, delimiter=",")
    next(reader)

    for row in reader:
        genre = row['GENRE_BOTA']
        espece = row['ESPECE']
        genre_espece = genre + ' ' + espece
        genre_espece = genre_espece.strip()
        espèces_différentes.append(genre_espece)

espèces_différentes

['',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 '',
 'Platanus',
 'Platanus',
 'Platanus',
 '',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 '',
 '',
 '',
 '',
 '',
 '',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 'Ginkgo biloba',
 'Salix',
 'Liriodendron tulipifera',
 'Liriodendron tulipifera',
 'Liriodendron tulipifera',
 'Catalpa bignonoides',
 'Platanus acerifolia',
 'Platanus acerifolia',
 'Platanus acerifolia',
 'Platanus acerifolia',
 'Platanus acerifolia',
 'Platanus acerifolia',
 'Platanus acerifolia',
 'Platanus acerifolia',
 'Platanus acerifolia',

In [8]:
for j in espèces_différentes:
    espèces_différentes.remove('')

espèces_différentes

ValueError: list.remove(x): x not in list

In [9]:
espèces_différentes

['Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Platanus',
 'Ginkgo biloba',
 'Salix',
 'Liriodendron tulipifera',
 'Liriodendron tulipifera',
 'Liriodendron tulipifera',
 'Catalpa bignonoides',
 'Platanus acerifolia',
 'Platanus acerifolia',
 'Platanus acerifolia',
 'Platanus acerifolia',
 'Platanus acerifolia',
 'Platanus acerifolia',
 'Platanus acerifolia',
 'Platanus acerifolia',
 'Platanus acerifolia',
 'Platanus acerifolia',
 'Platanus acerifolia',
 'Platanus acerifolia',
 'Platanus acerifolia',
 'Platanus acerifolia',
 'Pyrus',
 'Robinia pseudoacacia',

In [10]:
print(Counter(espèces_différentes).keys())

dict_keys(['Platanus', 'Ginkgo biloba', 'Salix', 'Liriodendron tulipifera', 'Catalpa bignonoides', 'Platanus acerifolia', 'Pyrus', 'Robinia pseudoacacia', 'Acer grosseri', 'Zelkova serrata', 'Parrotia persica', 'Fraxinus excelsior', 'Fagus sylvatica', 'Acer negundo', 'Magnolia grandiflora', 'Taxus baccata', 'Alnus cordata', 'Picea abies', 'Cupressus arizonica', 'Sequoiadendron giganteum', 'Pyrus calleryana', 'Pinus nigra', 'Betula verrucosa', 'Tilia platyphyllos', 'Acer pseudoplatanus', 'Tilia tomentosa', 'Acer platanoides', 'Cedrus deodara', 'Pinus sylvestris', 'Populus alba', 'Celtis australis', 'Tilia cordata', 'Populus nigra', 'Thuja', 'Sorbus aria', 'Sophora japonica', 'Acer cappadocicum', 'Acer campestre', 'Acer saccharinum', 'Liquidambar styraciflua', 'Prunus spinosa', 'Cedrus atlantica', 'Calocedrus decurrens', 'Pinus wallichiana', 'Prunus cerasus', 'Paulownia tomentosa', 'Tilia', 'Sorbus aucuparia', 'Gymnocladus dioicus', 'Ulmus minor', 'Corylus colurna', 'Prunus domestica', '

In [11]:
print("Compter le nombre d'espèces différentes: ", len(Counter(espèces_différentes)))

Compter le nombre d'espèces différentes:  345


In [12]:
print(Counter(espèces_différentes))

Counter({'Platanus acerifolia': 4422, 'Acer pseudoplatanus': 1347, 'Pinus nigra': 1304, 'Carpinus betulus': 1282, 'Acer platanoides': 1130, 'Tilia tomentosa': 1000, 'Acer campestre': 845, 'Liquidambar styraciflua': 806, 'Acer': 805, 'Fraxinus excelsior': 776, 'Betula verrucosa': 712, 'Pinus sylvestris': 567, 'Alnus cordata': 550, 'Acer saccharinum': 507, 'Liriodendron tulipifera': 506, 'Populus nigra': 493, 'Aesculus hippocastanum': 432, 'Pyrus calleryana': 430, 'Prunus serrulata': 425, 'Sophora japonica': 422, 'Chamaecyparis lawsoniana': 370, 'Betula utilis': 361, 'Gleditsia triacanthos': 350, 'Cercis siliquastrum': 344, 'Magnolia grandiflora': 330, 'Cedrus atlantica': 316, 'Robinia pseudoacacia': 284, 'Koelreuteria paniculata': 271, 'Fagus sylvatica': 267, 'Tilia platyphyllos': 266, 'Tilia cordata': 257, 'Zelkova serrata': 231, 'Quercus cerris': 225, 'Pinus': 222, 'Populus robusta': 221, 'Picea abies': 206, 'Taxus baccata': 202, 'Acer cappadocicum': 198, 'Alnus glutinosa': 184, 'Acer

### ● En utilisant la structure de données Dict compter le nombre de représentants de chaque genre botanique:

#### ○ Quel est le genre botanique le plus représenté?

In [13]:
Counter(list_genres)

Counter({'Acer': 5276,
         'Platanus': 4663,
         'Pinus': 2118,
         'Tilia': 1770,
         'Fraxinus': 1506,
         'Carpinus': 1296,
         'Betula': 1247,
         'Prunus': 996,
         'Populus': 966,
         'Alnus': 826,
         'Liquidambar': 809,
         'Quercus': 677,
         'Liriodendron': 507,
         'Pyrus': 501,
         'Aesculus': 459,
         'Sophora': 426,
         'Magnolia': 409,
         'Cedrus': 407,
         'Chamaecyparis': 373,
         'Gleditsia': 352,
         'Cercis': 351,
         'Robinia': 287,
         'Koelreuteria': 271,
         'Fagus': 267,
         'Malus': 262,
         'Picea': 252,
         'Zelkova': 233,
         'Ulmus': 210,
         'Cupressus': 203,
         'Taxus': 202,
         'Celtis': 193,
         'Corylus': 174,
         'Salix': 143,
         'Ginkgo': 111,
         'Crataegus': 108,
         'Paulownia': 101,
         'Morus': 100,
         'Cupressocyparis': 100,
         'Amélanchier': 96,
     

In [14]:
key_max = max(zip(Counter(list_genres).values(), Counter(list_genres).keys()))[1]
value_max = max(zip(Counter(list_genres).values(), Counter(list_genres).keys()))[0]
print(f"Le plus grand genre botanique compte est le {key_max} avec {value_max} entrées") 

Le plus grand genre botanique compte est le Acer avec 5276 entrées


#### ○ Listez tous les représentants uniques de leurs genre botanique

In [15]:
list_genres_uniques = []

for k, v in Counter(list_genres).items():

    list_genres_uniques.append(k)

list_genres_uniques

['Platanus',
 'Ginkgo',
 'Salix',
 'Liriodendron',
 'Catalpa',
 'Pyrus',
 'Robinia',
 'Acer',
 'Zelkova',
 'Parrotia',
 'Fraxinus',
 'Fagus',
 'Magnolia',
 'Taxus',
 'Alnus',
 'Picea',
 'Cupressus',
 'Sequoiadendron',
 'Pinus',
 'Betula',
 'Tilia',
 'Cedrus',
 'Populus',
 'Celtis',
 'Thuja',
 'Sorbus',
 'Sophora',
 'Liquidambar',
 'Prunus',
 'Calocedrus',
 'Paulownia',
 'Gymnocladus',
 'Ulmus',
 'Corylus',
 'Morus',
 'Clerodendron',
 'Gleditsia',
 'Libocedrus',
 'Quercus',
 'Lagerstroemia',
 'Taxodium',
 'Amélanchier',
 'Cercidiphyllum',
 'Ilex',
 'Malus',
 'Cedrela',
 'Photinia',
 'Tetradium',
 'Koelreuteria',
 'Halesia',
 'Crataegus',
 'Aesculus',
 'Ptelea',
 'Chimonanthus',
 'Carpinus',
 'Metasequoia',
 'Cercis',
 'Cornus',
 'Juglans',
 'Pteroceltis',
 'Broussonetia',
 'Phellodendron',
 'Cupressocyparis',
 'Cydonia',
 'Ficus',
 'Eleagnus',
 'Lonicera',
 'Albizia',
 'Pterocarya',
 'Davidia',
 'Buxus',
 'Cladastris',
 'Chionanthus',
 'Syringa',
 'Abies',
 'Sterculia',
 'Phillyrea',
 '

#### ○ Classer tous les genres botaniques en fonction de leur nombre (ordre décroissant)

In [16]:
Counter(list_genres)

Counter({'Acer': 5276,
         'Platanus': 4663,
         'Pinus': 2118,
         'Tilia': 1770,
         'Fraxinus': 1506,
         'Carpinus': 1296,
         'Betula': 1247,
         'Prunus': 996,
         'Populus': 966,
         'Alnus': 826,
         'Liquidambar': 809,
         'Quercus': 677,
         'Liriodendron': 507,
         'Pyrus': 501,
         'Aesculus': 459,
         'Sophora': 426,
         'Magnolia': 409,
         'Cedrus': 407,
         'Chamaecyparis': 373,
         'Gleditsia': 352,
         'Cercis': 351,
         'Robinia': 287,
         'Koelreuteria': 271,
         'Fagus': 267,
         'Malus': 262,
         'Picea': 252,
         'Zelkova': 233,
         'Ulmus': 210,
         'Cupressus': 203,
         'Taxus': 202,
         'Celtis': 193,
         'Corylus': 174,
         'Salix': 143,
         'Ginkgo': 111,
         'Crataegus': 108,
         'Paulownia': 101,
         'Morus': 100,
         'Cupressocyparis': 100,
         'Amélanchier': 96,
     

#### ○ Quelle est la proportion du genre botanique le plus représenté parmi tous les genres?

In [17]:
total = 0

for k, v in Counter(list_genres).items():

    total += v

value_max = max(zip(Counter(list_genres).values(), Counter(list_genres).keys()))[0]
key_max = max(zip(Counter(list_genres).values(), Counter(list_genres).keys()))[1]

proportion = round((value_max / total), 2)

print(f"Le genre botanique le plus représenté est {key_max}, dont la proportion dans le total est de {proportion}")

Le genre botanique le plus représenté est Acer, dont la proportion dans le total est de 0.17


In [18]:
total

30728

#### Quelle est la proportion de chacun des genres botaniques par rapport au total?

In [55]:
total = 0
proportions_genres = {}

for k, v in Counter(list_genres).items():

    total += v

for k, v in Counter(list_genres).items():

    prop = round((v / total), 5)

    if k not in proportions_genres:
        proportions_genres[k] = []
    proportions_genres[k].append(prop)

proportions_genres

{'Platanus': [0.15175],
 'Ginkgo': [0.00361],
 'Salix': [0.00465],
 'Liriodendron': [0.0165],
 'Catalpa': [0.0029],
 'Pyrus': [0.0163],
 'Robinia': [0.00934],
 'Acer': [0.1717],
 'Zelkova': [0.00758],
 'Parrotia': [0.00176],
 'Fraxinus': [0.04901],
 'Fagus': [0.00869],
 'Magnolia': [0.01331],
 'Taxus': [0.00657],
 'Alnus': [0.02688],
 'Picea': [0.0082],
 'Cupressus': [0.00661],
 'Sequoiadendron': [0.00046],
 'Pinus': [0.06893],
 'Betula': [0.04058],
 'Tilia': [0.0576],
 'Cedrus': [0.01325],
 'Populus': [0.03144],
 'Celtis': [0.00628],
 'Thuja': [0.00156],
 'Sorbus': [0.00286],
 'Sophora': [0.01386],
 'Liquidambar': [0.02633],
 'Prunus': [0.03241],
 'Calocedrus': [0.00013],
 'Paulownia': [0.00329],
 'Gymnocladus': [0.00029],
 'Ulmus': [0.00683],
 'Corylus': [0.00566],
 'Morus': [0.00325],
 'Clerodendron': [0.00016],
 'Gleditsia': [0.01146],
 'Libocedrus': [0.00153],
 'Quercus': [0.02203],
 'Lagerstroemia': [0.00163],
 'Taxodium': [0.00029],
 'Amélanchier': [0.00312],
 'Cercidiphyllum': 

In [56]:
prop_total = []

for k, v in Counter(proportions_genres).items():

    prop_total.append(v)

prop_total = list(itertools.chain.from_iterable(prop_total))

somme = 0

for j in prop_total:

    somme += j

round(somme, 5)

0.99997

In [57]:
with open('../data/proportion_genres.csv', 'w', newline='') as csv_file:

    header = ['Genre', 'Proportion']

    writer = csv.DictWriter(csv_file, fieldnames=header)

    writer.writeheader()

    for key in proportions_genres:
        writer.writerow({'Genre': key, 'Proportion': proportions_genres[key]})

#### ○ Les objectifs de la Métro sont-ils respectés en termes de diversité?

In [60]:
key_max = max(zip(Counter(espèces_différentes).values(), Counter(espèces_différentes).keys()))[1]
value_max = max(zip(Counter(espèces_différentes).values(), Counter(espèces_différentes).keys()))[0]
print(f"Le plus grand espèce compte est le {key_max} avec {value_max} entrées")

Le plus grand espèce compte est le Platanus acerifolia avec 4422 entrées


In [62]:
total = 0

for k, v in Counter(espèces_différentes).items():

    total += v

value_max = max(zip(Counter(espèces_différentes).values(), Counter(espèces_différentes).keys()))[0]
key_max = max(zip(Counter(espèces_différentes).values(), Counter(espèces_différentes).keys()))[1]

proportion = round((value_max / total), 2)

print(f"Le espèce botanique le plus représenté est {key_max}, dont la proportion dans le total est de {proportion}")

Le espèce botanique le plus représenté est Platanus acerifolia, dont la proportion dans le total est de 0.14


#### Il y a plus de 10% d'espèce Pltanus Acerifolia

#### mais, il n'y a pas plus de 20% d'un genre

In [63]:
total_esp = 0
proportions_especes = {}

for k, v in Counter(espèces_différentes).items():

    total_esp += v

for k, v in Counter(espèces_différentes).items():

    prop_esp = round((v / total_esp), 5)

    if k not in proportions_especes:
        proportions_especes[k] = []
    proportions_especes[k].append(prop_esp)

proportions_especes

{'Platanus': [0.00225],
 'Ginkgo biloba': [0.00361],
 'Salix': [0.00182],
 'Liriodendron tulipifera': [0.01647],
 'Catalpa bignonoides': [0.0029],
 'Platanus acerifolia': [0.14391],
 'Pyrus': [0.0001],
 'Robinia pseudoacacia': [0.00924],
 'Acer grosseri': [0.0001],
 'Zelkova serrata': [0.00752],
 'Parrotia persica': [0.00176],
 'Fraxinus excelsior': [0.02525],
 'Fagus sylvatica': [0.00869],
 'Acer negundo': [0.00576],
 'Magnolia grandiflora': [0.01074],
 'Taxus baccata': [0.00657],
 'Alnus cordata': [0.0179],
 'Picea abies': [0.0067],
 'Cupressus arizonica': [0.00195],
 'Sequoiadendron giganteum': [0.00026],
 'Pyrus calleryana': [0.01399],
 'Pinus nigra': [0.04244],
 'Betula verrucosa': [0.02317],
 'Tilia platyphyllos': [0.00866],
 'Acer pseudoplatanus': [0.04384],
 'Tilia tomentosa': [0.03254],
 'Acer platanoides': [0.03677],
 'Cedrus deodara': [0.00254],
 'Pinus sylvestris': [0.01845],
 'Populus alba': [0.00391],
 'Celtis australis': [0.00436],
 'Tilia cordata': [0.00836],
 'Populus 

### ● (plus dur) Comment la pratique de plantation a-t-elle évolué au cours du temps? Diversifie-t-on plus maintenant qu’avant ?

In [69]:
genre_bota_annee = []

with open('../data/trees.csv', newline="") as file:

    reader = csv.DictReader(file, delimiter=",")
    
    for row in reader:
        genre_bota_annee.append([row["ANNEEDEPLANTATION"], row["GENRE_BOTA"]])

genre_bota_annee

[['', ''],
 ['', ''],
 ['', ''],
 ['', ''],
 ['', ''],
 ['', ''],
 ['', ''],
 ['', ''],
 ['', ''],
 ['', ''],
 ['', ''],
 ['', ''],
 ['', ''],
 ['2017', ''],
 ['2017', ''],
 ['2017', ''],
 ['2017', ''],
 ['2017', 'Platanus'],
 ['2017', 'Platanus'],
 ['2017', 'Platanus'],
 ['2017', 'Platanus'],
 ['2017', 'Platanus'],
 ['2017', 'Platanus'],
 ['2017', 'Platanus'],
 ['2017', 'Platanus'],
 ['2017', 'Platanus'],
 ['2017', 'Platanus'],
 ['2017', 'Platanus'],
 ['2017', 'Platanus'],
 ['2017', 'Platanus'],
 ['2017', 'Platanus'],
 ['2017', 'Platanus'],
 ['2017', 'Platanus'],
 ['2017', ''],
 ['2017', 'Platanus'],
 ['2017', 'Platanus'],
 ['2017', 'Platanus'],
 ['2017', ''],
 ['2017', 'Platanus'],
 ['2017', 'Platanus'],
 ['2017', 'Platanus'],
 ['2017', 'Platanus'],
 ['2017', 'Platanus'],
 ['2017', ''],
 ['2017', ''],
 ['2017', ''],
 ['2017', ''],
 ['2017', ''],
 ['2017', ''],
 ['2017', 'Platanus'],
 ['2017', 'Platanus'],
 ['2017', 'Platanus'],
 ['2017', 'Platanus'],
 ['2017', 'Platanus'],
 ['2017', 

In [85]:
bota_genre_annee = []

with open('../data/trees.csv', newline="") as file:

    reader = csv.DictReader(file, delimiter=",")
    next(reader)

    for row in reader:
        bota_genre_annee.append([row["ANNEEDEPLANTATION"], row["GENRE_BOTA"]])

bota_genre_annee

[['', ''],
 ['', ''],
 ['', ''],
 ['', ''],
 ['', ''],
 ['', ''],
 ['', ''],
 ['', ''],
 ['', ''],
 ['', ''],
 ['', ''],
 ['', ''],
 ['2017', ''],
 ['2017', ''],
 ['2017', ''],
 ['2017', ''],
 ['2017', 'Platanus'],
 ['2017', 'Platanus'],
 ['2017', 'Platanus'],
 ['2017', 'Platanus'],
 ['2017', 'Platanus'],
 ['2017', 'Platanus'],
 ['2017', 'Platanus'],
 ['2017', 'Platanus'],
 ['2017', 'Platanus'],
 ['2017', 'Platanus'],
 ['2017', 'Platanus'],
 ['2017', 'Platanus'],
 ['2017', 'Platanus'],
 ['2017', 'Platanus'],
 ['2017', 'Platanus'],
 ['2017', 'Platanus'],
 ['2017', ''],
 ['2017', 'Platanus'],
 ['2017', 'Platanus'],
 ['2017', 'Platanus'],
 ['2017', ''],
 ['2017', 'Platanus'],
 ['2017', 'Platanus'],
 ['2017', 'Platanus'],
 ['2017', 'Platanus'],
 ['2017', 'Platanus'],
 ['2017', ''],
 ['2017', ''],
 ['2017', ''],
 ['2017', ''],
 ['2017', ''],
 ['2017', ''],
 ['2017', 'Platanus'],
 ['2017', 'Platanus'],
 ['2017', 'Platanus'],
 ['2017', 'Platanus'],
 ['2017', 'Platanus'],
 ['2017', 'Platanus']